accepted/2023/dotnet-nuget-setapikey.md

```diff
@@ -0,0 +1,234 @@
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+- `dotnet nuget apikey` commands `set` and `unset`
+
+The [`dotnet nuget config` spec](https://github.com/NuGet/Home/pull/12172) proposed `set` and `unset`, to mimic `git config` commands, rather than `add` and `remove` or `delete`, which commands like `dotnet add package` and `dotnet nuget add source` use.
+
+Some apps have a pre-defined environment variable name, rather than allowing customers to specify the environment variable name.
+For example, apps running on GitHub Actions expect to the a GitHub access token in the `GITHUB_TOKEN` environment variable.
+
+Some apps allow contents to be read from files by having the first character of the value be a `@`. NuGet does this with package source credentials, however, it has caused a significant amount of customer confusion because this `@filename` convention is not well known, and NuGet doesn't provide a clear error message. The GitHub CLI avoids this with its `gh api` commands by forcing customers to choose between `-f` which passes raw values, or `-F` which gets parsed, either as numbers or read from a file if it starts with a `@`.
+
+Several UNIX commands allow you to pass `-` which tells the command to read from STDIN. However, I don't believe this is compelling for `dotnet nuget apikey set`.
+This could be interesting to allow piped commands like `get-from-keyvault nuget_api_key | dotnet nuget apikey set -`.
+However, I believe that reading from environment variables and files are sufficient and customers will be able to work with it.
+
+- `nuget.config` support for unencrypted `<apikeys>`
+
+[`<packageSources>` already uses the `<add key="string" value="string">` syntax](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#packagesources), but it also adds a `protocolVersion` attribute.
+
+[`<packageSourceCredentials>` has a different schema](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#packagesourcecredentials).
+It still uses the `<add key="string" value="string">` syntax, allowing it to use `<add key="cleartextpassword" value="secret" />` and `<add> key="password" value="encrypted" />`.
+But that's because these elements are nested within an element whose name is the package source. To try to do the same with `<apikeys>` would require a breaking schema change, since currently it uses `key` as the package source and `value` as the API key.
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
+- Need to validate old nuget behavior when `apikeys` contains unexpected `encryption` element. Validate both read and write.
```

Comment:
On write: nuget.exe just overwrites the `<add/>` entry.
On read: nuget.exe just ignores the bool value.